aspnetcore/blazor/fundamentals/startup.md

```diff
@@ -0,0 +1,175 @@
+* Load compressed assemblies using an HTTP request and decompress them on the client for hosts that don't support fetching compressed contents from the server.
+* Alias resources to a different name by redirecting each `fetch` request to a new name.
+
+> [!NOTE]
+> External sources must return the required [Cross-Origin Resource Sharing (CORS)](https://www.w3.org/TR/cors/) headers for browsers to allow cross-origin resource loading. CDNs usually provide the required headers by default.
+
+`loadBootResource` parameters appear in the following table.
+
+| Parameter    | Description |
+| ------------ | ----------- |
+| `type`       | The type of the resource. Permissable types include: `assembly`, `pdb`, `dotnetjs`, `dotnetwasm`, and `timezonedata`. You only need to specify types for custom behaviors. Types not specified to `loadBootResource` are loaded by the framework per their default loading behaviors. |
+| `name`       | The name of the resource. |
+| `defaultUri` | The relative or absolute URI of the resource. |
+| `integrity`  | The integrity string representing the expected content in the response. |
+
+The `loadBootResource` function can return a URI string to override the loading process. In the following example, the following files from `bin/Release/net5.0/wwwroot/_framework` are served from a CDN at `https://cdn.example.com/blazorwebassembly/5.0.0/`:
+
+* `dotnet.*.js`
+* `dotnet.wasm`
+* Timezone data
+
+Inside the closing `</body>` tag of `wwwroot/index.html`:
+
+```html
+<script src="_framework/blazor.webassembly.js" autostart="false"></script>
+<script>
+  Blazor.start({
+    loadBootResource: function (type, name, defaultUri, integrity) {
+      console.log(`Loading: '${type}', '${name}', '${defaultUri}', '${integrity}'`);
+      switch (type) {
+        case 'dotnetjs':
+        case 'dotnetwasm':
+        case 'timezonedata':
+          return `https://cdn.example.com/blazorwebassembly/5.0.0/${name}`;
+      }
+    }
+  });
+</script>
+```
+
+To customize more than just the URLs for boot resources, the `loadBootResource` function can call `fetch` directly and return the result. The following example adds a custom HTTP header to the outbound requests. To retain the default integrity checking behavior, pass through the `integrity` parameter.
+
+Inside the closing `</body>` tag of `wwwroot/index.html`:
+
+```html
+<script src="_framework/blazor.webassembly.js" autostart="false"></script>
+<script>
+  Blazor.start({
+    loadBootResource: function (type, name, defaultUri, integrity) {
+      return fetch(defaultUri, { 
+        cache: 'no-cache',
+        integrity: integrity,
+        headers: { 'Custom-Header': 'Custom Value' }
+      });
+    }
+  });
+</script>
+```
+
+The `loadBootResource` function can return `null`/`undefined`, which results in the default loading behavior.
+
+Return a [`Response` promise](https://developer.mozilla.org/docs/Web/API/Response) for content created in JavaScript:
```

Comment:
I still don't think this section is helpful and might end up confusing readers since it's not a common pattern to create a `Response` object from scratch in JavaScript.

I would just say something like:

```
The `loadBootResource` function can return `null`/`undefined`, which results in the default loading behavior, or a [Response promise](https://developer.mozilla.org/docs/Web/API/Response)

Reply:
Ok ... I'll 🔪 it, but this is what the Brotli *Compression* example does.

Reply:
Perhaps, the link to the Brotli example should be part of that bullet. I'll do it that way. If they want to see it, all they have to do is follow the link.